rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1017,13 +1017,50 @@ private void logUpdateCollections(Transaction rskTx) {
         eventLogger.logUpdateCollections(sender);
     }
 
-    protected void processSVPFailure(Federation proposedFederation) {
+    protected void updateSvpState(Transaction rskTx) {
+        Optional<Federation> proposedFederationOpt = federationSupport.getProposedFederation();
+        if (proposedFederationOpt.isEmpty()) {
+            return;
+        }
+
+        // if the proposed federation exists and the validation period ended,
+        // we can conclude that the svp failed
+        Federation proposedFederation = proposedFederationOpt.get();
+        if (!validationPeriodIsOngoing(proposedFederation)) {
+            logger.info(
+                "[updateSvpState] Proposed federation validation failed at block {}. SVP failure will be processed",
+                rskExecutionBlock.getNumber()
+            );
+            processSVPFailure(proposedFederation);
+            return;
+        }
+
+        Keccak256 rskTxHash = rskTx.getHash();
+        // if the fund tx signed is present, then the fund transaction change was registered,
+        // meaning we can create the spend tx.
+        Optional<BtcTransaction> svpFundTxSigned = provider.getSvpFundTxSigned();
+        if (svpFundTxSigned.isPresent()) {
+            logger.info(
+                "[updateSvpState] Fund tx signed was found, so spend tx creation will be processed."
+            );
+            processSvpSpendTransactionUnsigned(rskTxHash, proposedFederation, svpFundTxSigned.get());
+            return;
+        }
+
+        // if the unsigned fund tx hash is not present, meaning we can proceed with the fund tx creation
+        Optional<Sha256Hash> svpFundTxHashUnsigned = provider.getSvpFundTxHashUnsigned();
+        if (svpFundTxHashUnsigned.isEmpty()) {
+            logger.info(
+                "[updateSvpState] Fund tx hash unsigned was not found, so fund tx creation will be processed."
+            );
+            processSvpFundTransactionUnsigned(rskTxHash, proposedFederation);
+        }
+    }
```

Comment:
What would happen if it reaches the end of this method? Will this be considered an error?

Reply:
nope, probably we are just waiting for the fund tx change to be registered

Reply:
Should we log that scenario?

Reply:
i don't think so. We won't be sure what the actual scenario is